src/core/public/application/utils.ts

```diff
@@ -119,3 +119,19 @@ const removeBasePath = (url: string, basePath: IBasePath, origin: string): strin
   }
   return basePath.remove(url);
 };
+
+export function getAppInfo(app: App<unknown> | LegacyApp): AppInfo | LegacyAppInfo {
+  if (isLegacyApp(app)) {
+    const { updater$, ...infos } = app;
+    return {
+      ...infos,
```

Comment:
I tried a generic approach to have proper return type depending on the input

```ts
function getAppInfo<T extends App<unknown> | LegacyApp>(
  app: T
): T extends App ? AppInfo : LegacyAppInfo {
  if (isLegacyApp(app)) {
    const { updater$, ...infos } = app;
    return {
      ...infos,
      legacy: true,
    };
  }
  const { updater$, mount, ...infos } = app;
  return {
    ...infos,
    legacy: false,
  };
}
```

However the  `isLegacyApp` typeguard was not enough for TS, and the return type was not correctly inferred, resulting in type errors. As this is only used to populate a `Map<string, AppInfo | LegacyAppInfo` anyway, I think current signature is still fine.

Reply:
I think it's fine too. Also there are only two remaining legacy apps right now, so we're close to gutting this anyways 😄